### PR TITLE
Prepend * to document title when tab hidden and new notifications

### LINKED ISSUE #892: Add current notification count to document title
I like the next-cloud app to be in a pinned tab in Firefox. Many apps get highlighted with a blue light when they have changed, EG a new notification or a new email arrived. 

Next cloud does not.

### Steps to reproduce
1. Pin tab in Firefox
2. Have a notification arrive
3. Observe the pinned tab does not hilight

### Expected behaviour
I would like the pinned tab to highlight when a notification arrives.

The notification appears when the document title changes.

Adding the current number of notifications to the document title would therefore make the pinned-tab changed-highlight work.

### Actual behaviour
The document title does not reflect the number of notifications.

The following patch does the work of making the notification-counts show up in the document title.

Presumably though some would prefer that it didn't, so it should have a setting somewhere? 

Where do we think would be best for such a setting?

I'd like to attach an emojii to each app/object_id instead of using the first two letters as it does here, but can't put the app svg images into the title description. Perhaps a setting should also allow associating an emojii with each object_id?

Just looking for some feedback on what kind of pull requests you might like for this.

[addNotifactionsToTitle.patch.txt](https://github.com/nextcloud/notifications/files/6151299/addNotifactionsToTitle.patch.txt)

Also would like the email app to send a notification for new emails, but one step at a time eh?

### PR DESCRIPTION
**Pull request by revpriest**
When a new notification arrives, if the tab is hidden, and we didn't
already change the title, we prepend a * to the title to make pinned
tabs show a changed-highlight and unpinned tabs show a visible change.

When the tab becomes visible again we revert the change, unless the
title was already changed by Talk or any other app.

Fix #892

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -228,6 +230,24 @@ export default {
 			return [h, s, l]
 		},
 
+		/**
+			 * Update the title to show * if there are new notifications
+			 * @param {Object} notifications The list of notifications
+			 */
+		async _updateDocTitleOnNewNotifications(notifications) {
+			if (notifications.length !== this._oldcount) {
```
[the remark this answers is not in the document]
Comment (revpriest) [reply src/App.vue:238]: Clearing any * added by the Notifications App if length decreases sounds okay, yeah.

Will take a look at the weekend.

Mostly need to know if I really should be committing the js/notifications-main.js.map and js/notifications-main.js files generated by the make.
 
I assume now that I should, but would be good to get confirmation of that.

Also no idea if I can edit a pull-request somehow or would need to submit a new one. Not really used the github pull-request before.

[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/App.vue:238]: You can do that, otherwise I will do it at the end.

**Comment by nickvergessen** [comment]
Fixed some minor coding style issues and rebased in https://github.com/nextcloud/notifications/pull/908 as I couldn't push to your repo.
I invited you to our org, so in the future you can push directly to the nextcloud repo and people can collaborate on your branch easily.

**Comment by revpriest** [comment]
Excellent, thank you.

**Comment by nickvergessen** [comment]
Well, thanks for you for the nice little improvement 😎

**Pull request closed without merging**